src/QsCompiler/Transformations/CallGraph.cs

```diff
@@ -54,6 +55,72 @@ internal static bool TryCombineTypeResolutionsForTarget(QsQualifiedName target,
             return success;
         }
 
+        /// <summary>
+        /// Reversed the dependencies of type parameters resolving to other type parameters in the given
+        /// dictionary to create a lookup whose keys are type parameters and whose values are all the type
+        /// parameters that can be updating by knowing the resolution of the lookup's associated key.
+        /// </summary>
+        private static ILookup<Tuple<QsQualifiedName, NonNullable<string>>, Tuple<QsQualifiedName, NonNullable<string>>> GetReplaceable(TypeParameterResolutions.Builder typeParamResolutions)
+        {
+             return typeParamResolutions
+                .Select(kvp => (kvp.Key, GetTypeParameters.Apply(kvp.Value)))   // Get any type parameters in the resolution type.
```

Comment:
Comments for every part of the query are very useful!

Reply:
LINQ queries can be a bit difficult to read sometimes. Especially this one. I thought it would be useful to add a little extra explanation. Glad to hear it helps!